### PR TITLE
Added missing Move Effect TODO tests - Volume B

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
2nd part of a long-term plan of adding all missing TODO tests to the repository.
~~Most of the effort of this one was due to Baton Pass 😆~~

Also:
- Fixed AI not considering `EFFECT_RAGING_BULL` similarly to `EFFECT_BRICK_BREAK`.
- Split Raging Bull tests into their own file.
- Renamed some Baton Pass-related tests so that their prefix becomes Baton Pass.
  - Eg. `Aqua Ring can be Baton Passed` becomes `Baton Pass passes Aqua Ring's effect`.
- Expanded Belch's test to show that it doesn't need to eat a berry before each use of the move.
- Unconfirmed interactions added to the specific files.
- Split `EFFECT_RAGING_BULL` from `EFFECT_BRICK_BREAK`'s file.
- Moved Grassy Terrain/Earthquake test to Earthquake file.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

**Comment by Pawkkie** [comment]
Oh my goodness I didn't even realize TODO tests were tracked

I will try to help with this effort, I LOVE the test system :)

(no relevance to this specific PR I just got excited lol)

**Comment by AsparagusEduardo** [comment]
> Oh my goodness I didn't even realize TODO tests were tracked
> 
> I will try to help with this effort, I LOVE the test system :)
> 
> (no relevance to this specific PR I just got excited lol)

Glad to see excitement to help the project :D
You can see everything that's missing in #2706

File: test/dynamax.c [hunk 1]
```diff
@@ -1158,6 +1158,8 @@ DOUBLE_BATTLE_TEST("(DYNAMAX) G-Max Terror traps both opponents")
     }
 }
 
+TO_DO_BATTLE_TEST("Baton Pass passes G-Max Terror's escape prevention effect");
```
Comment (AlexOn1ine) [inline test/dynamax.c:1161]:
```suggestion
TO_DO_BATTLE_TEST("(DYNAMAX) Baton Pass passes G-Max Terror's escape prevention effect");
```

File: test/dynamax.c [hunk 2]
```diff
@@ -1391,6 +1393,8 @@ DOUBLE_BATTLE_TEST("(DYNAMAX) G-Max Chi Strike boosts allies' crit chance")
     }
 }
 
+TO_DO_BATTLE_TEST("Baton Pass doesn't pass G-Max Chi Strike's effect");
```
Comment (AlexOn1ine) [inline test/dynamax.c:1396]:
```suggestion
TO_DO_BATTLE_TEST("(DYNAMAX) Baton Pass doesn't pass G-Max Chi Strike's effect");
```

File: test/battle/move_effect/substitute.c [hunk 3]
```diff
@@ -68,3 +68,5 @@ SINGLE_BATTLE_TEST("Substitute's HP cost doesn't trigger effects that trigger on
         NOT MESSAGE("Wobbuffet's Air Balloon popped!");
     }
 }
+
+TO_DO_BATTLE_TEST("Baton Pass passes Substitutes");
```
Comment (AlexOn1ine) [inline test/battle/move_effect/substitute.c:72]: Wouldn't it make sense for all Baton Pass tests to go into one file?

File: test/battle/move_effect/bulldoze.c [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#include "global.h"
+#include "test/battle.h"
+
+TO_DO_BATTLE_TEST("Bulldoze lowers the speed of every battler hit");
+TO_DO_BATTLE_TEST("Bulldoze's damage is halved when Grassy Terrain is in effect lowers the speed of every battler hit");
```
Comment (AlexOn1ine) [inline test/battle/move_effect/bulldoze.c:5]: There is no `EFFECT_BULLDOZE` anymore.

**Review by AlexOn1ine — COMMENTED** [review]
Since you've created a Baton Pass file why the decision to scatter them around other files as well?

**Comment by AsparagusEduardo** [comment]
> Since you've created a Baton Pass file why the decision to scatter them around other files as well?

Mainly to avoid making more giant test files.

Because Baton Pass has specific tests related to itself without influence of other moves/abilities (eg. Status1 and AI Ace mon tests), my intention is to keep those in the Baton Pass file, while other move interactions would stay in those specific moves' files.

File: test/battle/move_effect/bulldoze.c (hunk 4 above)
Comment (AsparagusEduardo) [reply test/battle/move_effect/bulldoze.c:5]: Forgot to delete this file once I realized this.

File: test/battle/move_effect/substitute.c (hunk 3 above)
Comment (AsparagusEduardo) [reply test/battle/move_effect/substitute.c:72]: See the main comment.

**Comment by AlexOn1ine** [comment]
> Because Baton Pass has specific tests related to itself without influence of other moves/abilities (eg. Status1 and AI Ace mon tests), my intention is to keep those in the Baton Pass file, while other move interactions would stay in those specific moves' files.

It is a bit misleading though since we are testing the Baton Pass interaction instead of status. We could also `PARAMETRIZE` it in this case

**Comment by AsparagusEduardo** [comment]
> It is a bit misleading though since we are testing the Baton Pass interaction instead of status. We could also `PARAMETRIZE` it in this case

Most interactions can be seen in as "A's interaction with B" or "B's interaction with A".
My main thought is that effects that interact with a lot of other effects should be spread because for example, you would have one massive file for *every* ability to check for Mold Breaker and such, so better to have them in their specific ability files.

**Comment by AlexOn1ine** [comment]
> > It is a bit misleading though since we are testing the Baton Pass interaction instead of status. We could also `PARAMETRIZE` it in this case
> 
> Most interactions can be seen in as "A's interaction with B" or "B's interaction with A". My main thought is that effects that interact with a lot of other effects should be spread because for example, you would have one massive file for _every_ ability to check for Mold Breaker and such, so better to have them in their specific ability files.

If an Ability is tested against Mold Breaker the outcome will be different since the result of each ability is different. 

In case of Baton Pass the question is just is Status passed or not which makes it easy to parametrize.

**Comment by AsparagusEduardo** [comment]
> In case of Baton Pass the question is just is Status passed or not which makes it easy to parametrize.

Ok, fair for the Status ones. I think I added the `Baton Pass doesn't pass non-volatile status conditions` forgetting that I added the individual ones in each status file. I'll delete those and just keep the single Status1 test in the Baton Pass file.

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**